controllers/argocd/hooks.go

```diff
@@ -22,6 +21,7 @@ func Register(h ...Hook) {
 	hooks = append(hooks, h...)
 }
 
+// nolint:unparam
```

Comment:
@chetan-rns I am slightly unsure why this is used. Can you please explain me ?

Reply:
Currently, golangci-lint throws an error because the `hint` parameter is not used anywhere. The comment will suppress the lint error since we want to keep the parameter. https://github.com/argoproj-labs/argocd-operator/pull/597#discussion_r837152358